utils/base_class.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+import requests
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.select import Select
+
+@pytest.mark.usefixtures("setup")
+class BaseClass:
+    def __init__(self, driver):
+        self.driver = driver
+
+    def verify_link_presence(self, text):
+        element = WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.LINK_TEXT, text)))
+
+    def no_errors_on_page(self):
```

Comment:
in its current implementation, this method only checks if the get response to the page url returns 200 response
in the browser, after this response is returned, JS starts working, and make more requests to fetch fonts, other JS code, or request something from the backend, and that's where 99% of errors come from.
So you need to either rename this method to smth like "validate_current_url_status", or change the logic to check network response log statuses

when initiating the driver, add these capabilities:

```
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
caps = DesiredCapabilities.CHROME
caps["goog:loggingPrefs"] = {"performance": "ALL"}
driver = webdriver.Chrome(desired_capabilities=caps)
```

and then when you want to validate all the response statuses:
```
    logs = driver.get_log("performance")
    for log_entry in logs:
        log_data = json.loads(log_entry["message"])["message"]
        if log_data["method"] == "Network.responseReceived":
            response = log_data["params"]["response"]
            url = response["url"]
            status = response["status"]
            assert status == 200, f"{url} response status code is {status}"
```